docs/extensibility/jsonata.md

```diff
@@ -3,27 +3,90 @@
 **Table of Contents**
 
 - [Overview](#overview)
+- [Scoping](#scoping)
 - [Preset functions](#preset-functions)
   - [_matchByLabelSelector_](#matchbylabelselectoritem-selectorpath)
   - [_matchEvents_](#matcheventsitem-kind-name)
   - [_compareStrings_](#comparestringsfirst-second)
 
 ## Overview
 
-This document describes how to use the preset JSONata functions.
+This document describes how to use jsonata expressions [JSONata](https://docs.jsonata.org/overview.html) throughout the extensions.
+
+## Scoping
+
+The primary data source of jsonata expressions changes depending on where it's used. Starting with the root it contains the whole resource, but whenever it's in a child whose parent has a **source** (in lists and details) or **path** (in forms) parameter, the scope changes to data from that source or path.
+
+Additionally in arrays, scope changes to the array item.
+
+For example for this resource:
+
+```yaml
+spec:
+  name: foo
+  description: bar
+  items:
+    - name: item-name
+      details:
+        status: ok
+```
+
+The following definition have their scope changed as follows:
+
+```yaml
+- source: spec.name # top level, scope is the same as resource
+
+- source: spec # top level, scope is the same as resource
+  children:
+    - source: name # parent has source=spec, therefore this refers to spec.name
+
+- children:
+    - source: spec.name # however there's no parent source here, therefore scope is still the resource
+
+- source: spec.items
+  children:
+    - source: name # parent data is an array, therefore scope changes to it's item - this refers to spec.items[0].name
+    - source: details.status # refers to spec.items[0].details.status (same as above)
+    - source: details # this changes scope for it's children again
+      children:
+        source: status # so this refers to spec.items[0].details.status
+```
+
+## Common variables
+
+Common variables are the primary means to bypass the default scoping.
+
+- **\$root** - always contains the reference to the resource, so any jsonata in the above can always be `$root.spec.name`
```

Comment:
```suggestion
- **\$root** - always contains the reference to the resource, so any JSONata can always be `$root.spec.name`.
```
In the above definition? example? Maybe we can omit the `in the above` part completely?